### PR TITLE
[k8s] Kong Kubernetes Infrastructure

### PR DESCRIPTION
**Pull request by Lewiky**
This PR adds the configuration required to deploy Kong in Kubernetes.

It's the first of several which will bring all of our Golang spec project into kube. 

It consists of several yaml files, each of which describe a required component of the k8s infra.

[This guide](https://kubernetes.io/docs/concepts/) may be helpful if you don't know anything about Kube.

### Test Plan

The `K8S-DEPLOYMENT.md` file specifies the steps needed to setup your environment to run the tests described here.

Once everything is running:

* Fetch the URLs into the service Minikube has assigned
```
minikube service kong --url
```

* Put these URLS and Port numbers into the `kong/configure-kong.sh` file

* Run the configure script

* Requests to configure the endpoints should succeed (Though pinging the endpoints will fail as the actual services don't exist yet)

### CODE REVIEW

**Review by c-harding — APPROVED** [review]
Nice³

File: K8s-DEPLOYMENT.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Kubernetes Deployment
+
+To deploy this project onto a kubernetes cluster follow these steps:
+
+## Cluster setup
+
+First, a Kubernetes cluster is required to deploy to. Instructions aren't provided here for production ready clusters, see the [Kubernetes Documentation](https://kubernetes.io/docs/tasks/) for that. 
+
+For local development, it's reccomended to use [Minikube](https://github.com/kubernetes/minikube). Follow the instructions to get it set up.
```
Comment (jaylees14) [inline K8s-DEPLOYMENT.md]:
```suggestion
For local development, it's recommended to use [Minikube](https://github.com/kubernetes/minikube). Follow the instructions to get it set up.
```

File: K8s-DEPLOYMENT.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+kubectl create -f kube/{DIR}
+```
+
+Will provision all of the resources described in that directory, so run that command for each subsection of infra you want to provision.
+
+As an example, to deploy all kong infrastructure:
+
+```
+kubectl create -f kube/kong
+```
```
Comment (jaylees14) [inline K8s-DEPLOYMENT.md:47]: Might be worth including something to bring down the cluster and clean up?

File: kube/kong/kong-db-service.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Service to expose the kong-db to the kong pod
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: kong-db
+  name: kong-db
+spec:
```
Comment (jaylees14) [inline kube/kong/kong-db-service.yaml:8]: Might be worth explicitly adding that this is `ClusterIP` in here - although it's the default, makes it easier to understand what's going on?

**Review by jaylees14 — APPROVED** [review]
Awesome stuff, nice description and pretty easy to follow! Couple of minor points just to perfect it, but really nice job 👍

**Pull request merged**